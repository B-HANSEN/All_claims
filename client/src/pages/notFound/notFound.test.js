import {
	render,
	screen,
	fireEvent,
	waitFor,
	waitForElement,
} from '@testing-library/react';
import ShallowRenderer from 'react-test-renderer/shallow';
import React from 'react';
import NotFound from '.';

const shallowRender = Component =>
	new ShallowRenderer(Component).getRenderOutput();

describe('Not Found Page tests', () => {
	it('should match snapshot', () => {
		expect(shallowRender(<NotFound />)).toMatchSnapshot();
	});

	it('should contain 404 elements', async () => {
		render(<NotFound />);
		const status = screen.getByText(/404/i);
		const notFoundPage = screen.getByText(/page not found./i);
		const notExisting = screen.getByText(/the page/i);
		const goHome = screen.getByText(/go home/i);

		expect(status).toBeInTheDocument();
		expect(notFoundPage).toBeInTheDocument();
		expect(notExisting).toBeInTheDocument();
		expect(goHome).toBeInTheDocument();
	});
});
