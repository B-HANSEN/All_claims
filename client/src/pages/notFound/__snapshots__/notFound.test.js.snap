// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Not Found Page tests should match snapshot 1`] = `null`;
